etc/ci-gcovr-coverage.py

```diff
@@ -0,0 +1,642 @@
+# data processing performed by lcov's geninfo command.  However, we
+# don't worry about parsing the *.gcna files, and backwards compatibility for
+# older versions of gcov is not supported.
+#
+# Outstanding issues
+#   - verify that gcov 3.4 or newer is being used
+#   - verify support for symbolic links
+#
+# For documentation, bug reporting, and updates,
+# see http://gcovr.com/
+#
+#  _________________________________________________________________________
+#
+#  Gcovr: A parsing and reporting tool for gcov
+#  Copyright (c) 2013 Sandia Corporation.
+#  This software is distributed under the BSD License.
+#  Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+#  the U.S. Government retains certain rights in this software.
+#  For more information, see the README.md file.
+# _________________________________________________________________________
+#
+# !! ** !! ** !!
+#
+# This file is entirely based on gcovr's main file, plus a json output
+# generator. I will try to get some changes pushed upstream
+#
+
+import locale
+import os
+import re
+import sys
+import json
+import mmap
+import hashlib
+
+from argparse import ArgumentParser, ArgumentTypeError
+from os.path import normpath
+from multiprocessing import cpu_count
+from tempfile import mkdtemp
+from shutil import rmtree
+
+from gcovr.gcov import get_datafiles, process_existing_gcov_file, process_datafile
+from gcovr.utils import (get_global_stats, build_filter, AlwaysMatchFilter,
+                    DirectoryPrefixFilter, Logger)
+from gcovr.version import __version__
+from gcovr.workers import Workers
+from gcovr.coverage import CoverageData
+
+# generators
+
+# count the number of lines in a file
+def mapcount(filename):
+    f = open(filename, "r+")
+    buf = mmap.mmap(f.fileno(), 0)
+    lines = 0
+    readline = buf.readline
+    while readline():
+        lines += 1
+    return lines
+
+# compute the md5 sum of a file
+def md5(fname):
+    hash_md5 = hashlib.md5()
+    with open(fname, "rb") as f:
+        for chunk in iter(lambda: f.read(4096), b""):
+            hash_md5.update(chunk)
+    return hash_md5.hexdigest()
+
+
+# TODO: branches?
+def write_coveralls_json(covdata, options):
+    report = {}
+    files = [] 
+    for file in covdata.keys():
+        data = covdata[file]
+        nlines = mapcount(file)
+        coverage = [ None ] * nlines
+
+        for line in data.all_lines:
+            hits = data.covered.get(line, 0)
+            if line - 1 < len(coverage):
+                coverage[line-1] = hits 
+            else:
+                # TODO: Log error
+                pass
+
+        files.append( { "name": options.root_filter.sub('', file)
+                    , "source_digest": md5(file)
+                    , "coverage": coverage } )
+
+    report["service_name"] = "travis-ci"
+    report["service_job_id"] = os.environ.get("TRAVIS_JOB_ID", "")
+    report["source_files"] = files
+    report["parallel"] = True
```

Comment:
The above is insufficient for our needs. But we are ignoring these keys in the merge script anyway, so perhaps just remove the previous 4 lines, to keep things simple?